src/Schema.js

```diff
@@ -363,6 +359,51 @@ class Schema {
     });
   }
 
+  validateNewClass(className, fields = {}, classLevelPermissions) {
+    if (this.data[className]) {
+      throw new Parse.Error(Parse.Error.INVALID_CLASS_NAME, `Class ${className} already exists.`);
+    }
+    if (!classNameIsValid(className)) {
+      return {
+        code: Parse.Error.INVALID_CLASS_NAME,
+        error: invalidClassNameMessage(className),
+      };
+    }
+    return this.validateSchemaData(className, fields, classLevelPermissions);
+  }
+
+  validateSchemaData(className, fields, classLevelPermissions) {
+    for (let fieldName in fields) {
+      if (!fieldNameIsValid(fieldName)) {
+        return {
+          code: Parse.Error.INVALID_KEY_NAME,
+          error: 'invalid field name: ' + fieldName,
+        };
+      }
+      if (!fieldNameIsValidForClass(fieldName, className)) {
+        return {
+          code: 136,
+          error: 'field ' + fieldName + ' cannot be added',
+        };
+      }
+      const error = fieldTypeIsInvalid(fields[fieldName]);
+      if (error) return { code: error.code, error: error.message };
+    }
+
+    for (let fieldName in defaultColumns[className]) {
+      fields[fieldName] = defaultColumns[className][fieldName];
+    }
+
+    let geoPoints = Object.keys(fields).filter(key => fields[key] && fields[key].type === 'GeoPoint');
+    if (geoPoints.length > 1) {
+      return {
+        code: Parse.Error.INCORRECT_TYPE,
+        error: 'currently, only one GeoPoint field may exist in an object. Adding ' + geoPoints[1] + ' when ' + geoPoints[0] + ' already exists.',
```

Comment:
No, I think the adapter should return an error when you try to update the schema in a way that the adapter doesn't support. Then there is one less function that the adapter needs to implement.